src/methods/clipping/difference.jl

```diff
@@ -162,6 +162,7 @@ function _difference(
     local polys
     for (i, poly_b) in enumerate(GI.getpolygon(multipoly_b))
         polys = difference(i == 1 ? multipoly_a : GI.MultiPolygon(polys), poly_b; target, fix_multipoly)
```

Comment:
We should split out this conditional into an is/else, this logic is a bit hard to follow.

Reply:
Maybe with a few comments... Given the bug and anshuls comment

Reply:
Updated!